app/src/main/java/org/oppia/android/app/activity/route/ActivityRouter.kt

```diff
@@ -18,19 +18,21 @@ class ActivityRouter @Inject constructor(
   /** Opens the activity corresponding to the specified [destinationScreen]. */
   fun routeToScreen(destinationScreen: DestinationScreen) {
     destinationRoutes[destinationScreen.destinationScreenCase]?.let { route ->
-      when (destinationScreen.destinationScreenCase) {
-        DestinationScreenCase.RECENTLY_PLAYED_ACTIVITY_PARAMS -> {
-          activity.startActivity(
-            route.createIntent(
-              activity,
-              destinationScreen
-            )
+      if (destinationScreen.destinationScreenCase ==
```

Comment:
Why switch to an if check? The ``when`` was much cleaner.

Reply:
I shift to if because it was giving me warning as I have shifted DestinationScreenCase.DESTINATIONSCREEN_NOT_SET outside when.

Reply:
What's the warning, though? It's not obvious why this change is needed, and it's actually not idiomatic to Kotlin or what we do elsewhere in the codebase for similar patterns. I think there needs to be a compelling reason why we're going against convention here.